--- a/src/getopt.h
+++ b/src/getopt.h
@@ -67,11 +67,11 @@ struct opt_entry {
  * Processes all the command line options. Updates the supplied flags and
  * parameters specified in the options list.
  */
-extern void opt_process(struct opt_entry *list, opt_flags *flg, char **argv);
+extern void opt_process(struct opt_entry *list, opt_flags *flg, int argc, char **argv);
 
 /*
  * Checks option dependencies.
  */
-extern void opt_check(struct opt_entry *list, opt_flags flg, char **argv);
+extern void opt_check(struct opt_entry *list, opt_flags flg, int argc, char **argv);
 
 #endif
