Index: sched/sched_shmem.h
===================================================================
--- sched/sched_shmem.h (revision 24231)
+++ sched/sched_shmem.h (working copy)
@@ -32,9 +32,9 @@
 // Increase as needed
 //
 #define MAX_PLATFORMS       50
-#define MAX_APPS            10
-#define MAX_APP_VERSIONS    50
-#define MAX_ASSIGNMENTS     10
+#define MAX_APPS            35
+#define MAX_APP_VERSIONS    250
+#define MAX_ASSIGNMENTS     35

 // Default number of work items in shared mem.
 // You can configure this in config.xml (<shmem_work_items>)
