Index: Makefile
===================================================================
--- Makefile	(revision 400)
+++ Makefile	(working copy)
@@ -5,7 +5,7 @@
 #TOOLSPREFIX=i586-mingw32msvc-
 
 LOCALINC=-I/usr/local/include
-LOCALLIB=-L/usr/local/lib
+LOCALLIB=-L/usr/local/lib -L../../boinc-libs -L../../../../boinc-libs
 
 CC=$(TOOLSPREFIX)gcc
 CPP=$(TOOLSPREFIX)g++
@@ -25,15 +25,18 @@
 	@echo "	pentium4              	 Intel Pentium 4"
 	@echo "	prescott                 Intel Pentium 4 with SSE3"
 	@echo "	pentium-m              	 Intel Pentium M"
-	@echo "	athlon			 AMD Athlon (k7)"
-	@echo "	athlon-xp		 AMD Athlon XP (k7) with SSE"
+	@echo "	athlon                   AMD Athlon (k7)"
+	@echo "	athlon-xp                AMD Athlon XP (k7) with SSE"
+	@echo "	k8                       AMD Athlon (k8)"
+	@echo " native			 Native Architecture for corei7"
+	@echo "	barcelona                AMD Phenom (barcelona) with SSE4a"
 	@echo "	x86_64                   AMD Opteron/Athlon64 (k8)"
 	@echo "	nocona                   Intel 64-bit-capable Xeon/Pentium"
 	@echo "	ppc_970             	 PowerPC 970"
 	@echo "	ppc_7450             	 PowerPC 7450"
-	@echo "	doc			 Documentation"
+	@echo "	doc                      Documentation"
 	@echo "	snapshot       		 Sources snapshot"
-	@echo "	install			 Installation"
+	@echo "	install                  Installation"
 	@echo "	clean         		 Clean up"
 
 pentium2 :
@@ -57,6 +59,15 @@
 athlon-xp :
 	@ARCH="athlon-xp" $(MAKE) x86common
 
+k8 :
+	@ARCH="k8" $(MAKE) common
+
+barcelona :
+	@ARCH="barcelona" $(MAKE) common
+
+native : 
+	@ARCH="native" $(MAKE) x86common
+
 x86_64 :
 	@ARCH="k8" $(MAKE) common
 
