Index: gnfs/poly/stage2/root_sieve_line.c
===================================================================
--- gnfs/poly/stage2/root_sieve_line.c	(revision 639)
+++ gnfs/poly/stage2/root_sieve_line.c	(working copy)
@@ -343,7 +343,7 @@
 	uint32 num_primes = rs->num_primes;
 	uint16 *block = rs->sieve_block;
 
-	printf("%u ", num_blocks); fflush(stdout);
+	//printf("%u ", num_blocks); fflush(stdout);
 
 	if (mpz_cmp_ui(x->mp_lattice_size, 1) == 0)
 		prepare_sieve_line(rs);
