Index: index.php
===================================================================
--- index.php	(revision 160)
+++ index.php	(revision 161)
@@ -110,6 +110,7 @@
     <table border=\"0\"><tr valign=\"top\">
     <td><b>Browse database:</b>
     <ul> 
+        <li><a href=\"status.php\">Result Dashboard</a></li>
         <li><a href=\"db_form.php?table=result&amp;detail=low\">Results</a></li>
         <li><a href=\"db_form.php?table=workunit\">Workunits</a></li>
         <li><a href=\"db_form.php?table=host&amp;detail=low\">Hosts</a></li>
